subsys/net/ip/ipv4.c

```diff
@@ -141,41 +141,30 @@ enum net_verdict net_ipv4_process_pkt(struct net_pkt *pkt)
 		goto drop;
 	}
 
+	if (!net_ipv4_is_my_addr(&hdr->dst) &&
```

Comment:
Can you update the commit message to say "All IPv4 destination address related drop reasons ...", because that's what happens here.